### ISSUE TITLE
Can't find the code about the demultiplex part using key vectors

### ISSUE THREAD

**Issue by LuLuLuyi** [princeton-nlp/datamux-pretraining#1]
This is a wonderful work after DataMux, the multiplexer and demultiplexer are very interesting. But I have a problem about the demultiplexer part after reading your paper and code.
In Mux-PLMs, to address the problem of the demultiplexer in DataMUX which requires a prefix. N (private) key vectors are proposed to be used to demultiplex the output representation.
However, I can't find the code about the the demultiplex part using N (private) key. I found some code about demultiplex part in multiplexing_pretraining_bert.py and pasting the part I found as follows. It seems that prefix in DataMUX is used to demultiple instead of N (private) key vectors.
`if self.demuxing_variant == "index":

            # add the prefix
            # [CLS1, <s>, <s>, <s>, <s>]
            # [<s>, CLS2, <s>, <s>, <s>]
            # [<s>, <s>, CLS3, <s>, <s>]
            # [<s>, <s>, <s>, CLS4, <s>]
            # [<s>, <s>, <s>, <s>, CLS5]
            # let us just assume the last 5 tokens barring the masked token
            # are the cls tokens (easiest way to make use of existing vocab)

            # prefix 5 x 5

            prefix = torch.full(
                (num_instances, num_instances), 680, device=input_ids.device
            )
            prefix[
                torch.arange(num_instances, device=input_ids.device),
                torch.arange(num_instances, device=input_ids.device),
            ] = (
                -(torch.arange(num_instances, device=input_ids.device) + 2) + 680
            )

            # [-2   <s>, <s>, <s>, <s>]
            # [<s>, -3, <s>, <s>, <s>]
            # [<s>, <s>, -4, <s>, <s>]
            # [<s>, <s>, <s>, -5, <s>]
            # [<s>, <s>, <s>, <s>, -6]
            # +  size of vocab
            cls_tokens = torch.full((num_instances, 1), 101, device=input_ids.device)
            prefix = torch.cat([prefix, cls_tokens], dim=1)

            prefix = prefix.repeat(modified_batch_size, 1)
            input_ids = input_ids[: (modified_batch_size * num_instances)]
            input_ids = torch.cat([prefix, input_ids[:, : -(num_instances + 1)]], dim=1)
            modified_seq_length = seq_length
            special_tokens_end_position = num_instances + 1

        elif self.demuxing_variant == "index_pos":
            cls_tokens = torch.full((num_instances, 1), 101, device=input_ids.device)
            cls_tokens = cls_tokens.repeat(modified_batch_size, 1)
            input_ids = input_ids[: (modified_batch_size * num_instances)]
            input_ids[:, 0:1] = cls_tokens
            modified_seq_length = seq_length
            special_tokens_end_position = 0`
I'm not sure if I miss something, I want to find the code of  initializing N (private) key vectors.
Thanks!
